### DISCUSSION TITLE
[Question] How to connect custom Spanner hosts without modifying the Driver Settings

### DISCUSSION THREAD

**Discussion by h6ah4i** [dbeaver/dbeaver#21576] [Q&A]
### Description

I want a equivalent feature of the `SPANNER_EMULATOR_HOST` environment variable for **independently configurable with each connections**.
https://cloud.google.com/spanner/docs/use-oss-jdbc#connect-emulator

I know that I can connect to custom emulator endpoint with modifying the `URL Template` property in the Driver Settings. However, the Driver Settings parameter is shared with all connections.

### DBeaver Version

Community Edition 23.2.2.202310081633

### Operating System

macOS Sonoma 14.0

### Database and driver

Database:
- Multiple Spanner emulators running on Docker containers

Driver name: Google Cloud Spanner

**Comment by LonwoLonwo** [comment]
Hello @h6ah4i 

If I understood your question correctly, you can use the [Driver Manager](https://github.com/dbeaver/dbeaver/wiki/Database-drivers#adding-driver-configuration-in-dbeaver).

If you create a Copy of your Spanner driver, all changes in this copy will be enabled only for the copy version.

![2023-10-18 10_27_46-DBeaver Ultimate 23 2 3 - Sessions - localhost 21](https://github.com/dbeaver/dbeaver/assets/45152336/ed98ecac-c4b4-47da-9fc9-3fec6c2aeeb1)

**Comment by h6ah4i** [comment]
@LonwoLonwo Thank you for your reply. The driver manager saved me 👍
